src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -0,0 +1,223 @@
+      );
+    return {
+      href,
+      name,
+      key: index.toString(),
+      icon: <EuiIcon type="stopFilled" color={workspace.color ?? 'primary'} />,
+    };
+  };
+
+  const getWorkspaceListItems = () => {
+    const workspaceListItems = filteredWorkspaceList.map((workspace, index) =>
+      workspaceToItem(workspace, index)
+    );
+    const length = workspaceListItems.length;
+    workspaceListItems.push({
+      icon: <EuiIcon type="plus" />,
+      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.createWorkspace', {
+        defaultMessage: 'Create workspace',
+      }),
+      key: length.toString(),
+      href: formatUrlWithWorkspaceId(
+        getUrlForApp(WORKSPACE_CREATE_APP_ID, {
+          absolute: false,
+        }),
+        currentWorkspace?.id ?? '',
+        basePath
+      ),
+    });
+    workspaceListItems.push({
+      icon: <EuiIcon type="folderClosed" />,
+      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.allWorkspace', {
+        defaultMessage: 'All workspaces',
+      }),
+      key: (length + 1).toString(),
+      href: formatUrlWithWorkspaceId(
+        getUrlForApp(WORKSPACE_LIST_APP_ID, {
+          absolute: false,
+        }),
+        currentWorkspace?.id ?? '',
+        basePath
+      ),
+    });
+    return workspaceListItems;
+  };
+
+  const currentWorkspaceButton = (
+    <EuiCollapsibleNavGroup>
+      <EuiFlexGroup>
+        <EuiFlexItem>
+          <EuiIcon type="logoOpenSearch" size="l" />
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiText>
+            <strong> {currentWorkspaceName} </strong>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem>
+          <EuiIcon type="arrowDown" onClick={onButtonClick} />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiCollapsibleNavGroup>
+  );
+
+  const currentWorkspaceTitle = (
+    <EuiFlexGroup>
+      <EuiFlexItem>
+        <EuiIcon type="logoOpenSearch" size="l" />
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiText>
+          <strong> {currentWorkspaceName} </strong>
+        </EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem>
+        <EuiIcon type="cross" onClick={closePopover} />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+
+  const panels = [
+    {
+      id: 0,
+      title: currentWorkspaceTitle,
+      items: [
+        {
+          name: (
+            <EuiText>
+              <strong>
+                {i18n.translate('core.ui.primaryNav.workspacePickerMenu.workspaceList', {
+                  defaultMessage: 'Workspaces',
+                })}
+              </strong>
+            </EuiText>
+          ),
+          icon: 'folderClosed',
+          panel: 1,
+        },
+        {
+          name: managementWorkspaceName,
+          icon: 'managementApp',
+          href: formatUrlWithWorkspaceId(
+            getUrlForApp(WORKSPACE_OVERVIEW_APP_ID, {
+              absolute: false,
+            }),
+            MANAGEMENT_WORKSPACE,
```

Comment:
Can we rename `PUBLIC_WORKSPACE` and `MANAGEMENT_WORKSPACE` to `PUBLIC_WORKSPACE_ID` and `MANAGEMENT_WORKSPACE_ID`? From the name, these two constants might be considered as a workspace object instead of workspace id.

Reply:
Sounds good, I can add it to this PR